src/file.ts

```diff
@@ -1781,6 +1781,9 @@ class File extends ServiceObject<File> {
       });
     });
 
+    // forwards errors
+    fileWriteStream.on('error', stream.emit.bind(stream, 'error'));
```

Comment:
This line is above a bit:

```js
const stream = streamEvents(
  pumpify([
    gzip ? zlib.createGzip() : through(),
    validateStream,
    fileWriteStream,
  ])
) as Duplex;
```

`pumpify` should be catching errors on all of these streams and emitting it for us on `stream` already. I put together a little recreation here:

```js
function createWriteStream() {
  const stream = require('stream')
  const pumpify = require('pumpify')

  // Mocking streams
  const gzipStream = new stream.Stream()
  const validateStream = new stream.Stream()
  const fileWriteStream = new stream.Stream()

  // Simulate an error
  setTimeout(() => fileWriteStream.emit('error', new Error('Error message')), 1000)

  return pumpify([gzipStream, validateStream, fileWriteStream])
}

createWriteStream().on('error', err => console.log('Oh no!', err))
// Oh no! Error: Error message
```

Reply:
you are right, thanks for the example code.. I was on the wrong path ;)
 I debugged it further, see issue #833, closing this PR therefore.